internal/services/recoveryservices/recovery_services_vault_resource.go

```diff
@@ -52,6 +53,45 @@ func resourceRecoveryServicesVault() *pluginsdk.Resource {
 
 			"resource_group_name": azure.SchemaResourceGroupName(),
 
+			"encryption": {
+				Type:         pluginsdk.TypeList,
+				Optional:     true,
+				RequiredWith: []string{"identity"},
+				MaxItems:     1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"key_id": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: keyvaultValidate.NestedItemIdWithOptionalVersion,
+						},
+						"infrastructure_encryption_state": {
```

Comment:
could we change this to a bool?
```suggestion
						"infrastructure_encryption_enabled": {
```

Reply:
Roger that @katbyte !